boskos/cmd/cli/cli.go

```diff
@@ -284,15 +285,21 @@ Examples:
 			}()
 
 			tick := time.Tick(options.heartbeat.period)
+			// transient network issues should not cause us to stop trying
+			var failures int
 			work := func() bool {
 				select {
 				case <-tick:
 					if err := options.c.Update(resource.Name, resource.State, resource.UserData); err != nil {
 						fmt.Fprintf(cmd.ErrOrStderr(), "failed to send heartbeat for resource %q: %v\n", resource.Name, err)
-						exit(1)
-						return true
+						failures++
+						if failures > options.heartbeat.retries {
+							exit(1)
+							return true
+						}
+					} else {
+						fmt.Fprintf(cmd.OutOrStdout(), "heartbeat sent for resource %q\n", resource.Name)
```

Comment:
Would it make sense to set `failures` back to 0 on success?

Reply:
Could do that as well, too. I was just looking for a simple approach :)

Reply:
+1